File: src/AsArrayAsList.java
import java.util.Arrays;
import java.util.List;

/**
 * This is the type AsArrayAsList.
 *
 * @author <a href="mailto:jsvg88@scania.com">jsvg88</a>
 * @since 2017-09-28
 */
public class AsArrayAsList {
    public static void main(String... args) {

        final List<String> strings = Arrays.asList("Hello world".split("(?!^)"));
        final String[] strings2 = (String[]) strings.toArray();
        final List<String> strings3 = Arrays.asList(strings2);
        final String[] strings4 = (String[]) strings3.toArray();

        System.out.println(Boolean.valueOf(strings == strings3));
        System.out.println(Boolean.valueOf(strings2 == strings4));


        System.out.println(Arrays.asList(strings4));


    }
}

File: src/AsArrayAsList.kt
/**
 * This is the type AsArrayAsList.
 *
 * @author <a href="mailto:jsvg88@scania.com">jsvg88</a>
 * @since 2017-09-28
 */
import java.util.Arrays

/**
 * This is the type AsArrayAsList.

 * @author [jsvg88](mailto:jsvg88@scania.com)
 * *
 * @since 2017-09-28
 */
    fun main(args: Array<String>) {

        val strings = Arrays.asList(*"Hello world".split("(?!^)".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray())
        val strings2 = strings.toTypedArray()
        val strings3 = Arrays.asList(*strings2)
        val strings4 = strings3.toTypedArray()

        println(java.lang.Boolean.valueOf(strings === strings3))
        println(java.lang.Boolean.valueOf(strings2 === strings4))


        println(Arrays.asList(*strings4))


    }



